tools/delete.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import json
+import sys
+
+from core_data_modules.logging import Logger
+from id_infrastructure.firestore_uuid_table import FirestoreUuidTable
+from storage.google_cloud import google_cloud_utils
+
+
+def query_yes_no(question, default="yes"):
+    """Asks a yes/no question via input() and returns user input.
+    """
+    valid = {"yes": True, "y": True, "ye": True, "no": False, "n": False}
+    if default is None:
+        prompt = " [y/n] "
+    elif default == "yes":
+        prompt = " [Y/n] "
+    elif default == "no":
+        prompt = " [y/N] "
+    else:
+        raise ValueError(f"invalid default answer: {default}")
+
+    while True:
+        sys.stdout.write(question + prompt)
+        choice = input().lower()
+        if choice == "" and default is not None:
+            return valid[default]
+        elif choice in valid:
+            return True
+        else:
+            sys.stdout.write("Please respond with 'yes' or 'no' " "(or 'y' or 'n').\n")
+
+
+def concat_description(desc_list):
+    if len(desc_list) == 0:
+        return ''
+    if len(desc_list) == 1:
+        return desc_list[0]
+    if len(desc_list) == 2:
+        return desc_list[0] + ' and ' + desc_list[1]
+    return ', '.join(desc_list[:-1]) + ', and ' + desc_list[-1]
+
+
+def init_uuid_table_client(google_cloud_credentials_file_path, firebase_credentials_file_url, firebase_table_name):
+    log.info("Initialising uuid table client...")
+    credentials = json.loads(google_cloud_utils.download_blob_to_string(
+        google_cloud_credentials_file_path, firebase_credentials_file_url
+    ))
+
+    uuid_table = FirestoreUuidTable.init_from_credentials(credentials, firebase_table_name, "avf-participant-uuid-")
+    log.info(f"Initialised {firebase_table_name} uuid table client")
+
+    return uuid_table
+
+
+log = Logger(__name__)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Deletes data from Firestore")
+    parser.add_argument("--startswith", metavar="startswith",
+                        help="Enter prefix to be searched for in the data")
+    parser.add_argument("--contains", metavar="contains",
+                        help="Enter substring to be searched for in the data")
+    parser.add_argument("--endswith", metavar="endswith",
+                        help="Enter suffix to be searched for in the data")
```

Comment:
What was the logic behind using these particular filters? 

If you want flexibility, would it make more sense to use a regex? e.g. I can imagine it would be useful to say "doesn't contain :, or doesn't start with tel or telegram"

Reply:
Thanks, this has been helpful.